Basic_viewer/doc/Basic_viewer/CGAL/Qt/Basic_viewer_qt.h

```diff
@@ -0,0 +1,256 @@
+
+  /// clears the basic viewer, i.e., remove all its elements.
+  void clear();
+
+  /// returns `true` if the viewer is empty.
+  bool empty() const;
+
+  /// returns the bounding box of all the elements in the viewer.
+  const CGAL::Bbox_3& bounding_box() const;
+
+  /// returns `true` if the clipping plane is enabled.
+  bool clipping_plane_enabled() const;
+
+  /// returns the clipping plane when it is enabled.
+  CGAL::Exact_predicates_inexact_constructions_kernel::Plane_3 clipping_plane() const;
+
+  /// adds the given point in the viewer.
+  template <typename KPoint>
+  void add_point(const KPoint &p);
+
+  /// adds the given colored point in the viewer.
+  template <typename KPoint>
+  void add_point(const KPoint &p, const CGAL::IO::Color &color);
+
+  /// adds the given segment in the viewer.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2);
+
+  /// adds the given colored segment in the viewer.
+  template <typename KPoint>
+  void add_segment(const KPoint &p1, const KPoint &p2,
+                   const CGAL::IO::Color &color);
+
+  /// adds the given ray in the viewer: a half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v);
+
+  /// adds the given colored ray in the viewer: an half line starting from `p` and having `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_ray(const KPoint &p, const KVector &v,
+               const CGAL::IO::Color &color);
+
+  /// adds the given line in the viewer, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v);
+
+  /// adds the given colored line in the viewer, defined by `p` and `v` as direction.
+  template <typename KPoint, typename KVector>
+  void add_line(const KPoint &p, const KVector &v,
+                const CGAL::IO::Color &color);
+
+  /// starts a new face.
+  void face_begin();
+
+  /// starts a new colored face.
+  void face_begin(const CGAL::IO::Color &color);
+
+  /// returns `true` iff a face is started.
+  bool a_face_started() const;
+
+  /// adds the given point in the current face.
+  /// @pre `a_face_started()`
+  template <typename KPoint> bool add_point_in_face(const KPoint &kp);
+
+  /// adds the given point in the current face, having the vertex normal.
+  /// @pre `a_face_started()`
+  template <typename KPoint, typename KVector>
+  bool add_point_in_face(const KPoint &kp, const KVector &p_normal);
+
+  /// ends the current face.
+  /// @pre `a_face_started()`
+  void face_end();
+
+  /// adds the given text at the given position in the viewer.
+  template <typename KPoint>
+  void add_text(const KPoint &kp, const char *txt);
+
+  /// adds the given text at the given position in the viewer.
+  template <typename KPoint>
+  void add_text(const KPoint &kp, const std::string &txt);
+
```

Comment:
I do not get why those non-const methods are duplicated between the viewer and the scene. It seems a mistake. Probably due to historical reasons, but as it is it looks like a design flaw. That is also probably why the graphics scene is taken by refence to non-const.

Reply:
All these methods are now removed from the viewer.